tf2_yolov4/model.py

```diff
@@ -69,13 +83,16 @@ def YOLOv4(
     medium_features = neck(lower_features)
     upper_features = head(medium_features)
 
-    return tf.keras.Model(inputs=inputs, outputs=upper_features, name="YOLOv4")
+    yolov4 = tf.keras.Model(inputs=inputs, outputs=upper_features, name="YOLOv4")
+    if weights is None:
+        return yolov4
 
+    if weights == "darknet":
+        if not is_darknet_weights_available():
+            download_darknet_weights(yolov4)
 
-if __name__ == "__main__":
-    model = YOLOv4(input_shape=(608, 416, 3), num_classes=80, anchors=YOLOV4_ANCHORS)
+        yolov4.load_weights(DARKNET_AS_H5_PATH, by_name=True, skip_mismatch=True)
```

Comment:
I think the logic of check if weights are available before downloading it does not belong here (the model). It should e in the weights.py file. For instance, there could be a function `get_available_weights(weights="darknet")` in weights.py that : 

1. checks if the args is correct i.e., in the list ["darknet"], raise an error otherwise
2. check if the dir and weights file exists (`is_darknet_weights_available()`)
3. if not download it (`download_darknet_weights()`)
4. in the end, return the file path that you load using `yolov4.load_weights(weights_paht)`

This way, there is no weights file management in the model file, and a single import from weights.py